app/scripts/views/mixins/avatar-mixin.js

```diff
@@ -84,13 +94,15 @@ define([
     },
 
     updateProfileImage: function (profileImage) {
-      var account = this.getSignedInAccount();
+      var self = this;
+      var account = self.getSignedInAccount();
       account.setProfileImage(profileImage);
-      this.user.setAccount(account);
-
-      this.notifications.profileChanged({
-        uid: account.get('uid')
-      });
+      return self.user.setAccount(account)
+        .then(function () {
+          self.notifications.profileUpdated({
```

Comment:
These three lines are duplicated in `updateDisplayName`. Seems like a candidate for its own function in case `profileUpdated` needs to be updated in the future.